### ISSUE TITLE
[a11y] Cannot select DatePicker year/month dropdown when voice assist enabled

### ISSUE THREAD

**Issue by ChristianEdwardPadilla** [flutter/flutter#141992]
Internal: b/244681884

### Steps to reproduce

1. Run sample code.
2. Enable voice assist (VoiceOver, TalkBack, etc.).
3. Activate the "Select Date" button, which launches the calendar widget.
4. Attempt to interact with the month/year dropdown (currently shows as "February 2025" with a down arrow icon).

### Expected results

It is possible to focus the month/year dropdown.

### Actual results

It is not possible to focus the month/year dropdown.

### Discussion

I was able to repro this issue on Android, iOS, and Web (Chrome on macOS). I did not try other platforms.

I believe this is separate from https://github.com/flutter/flutter/issues/121602.

### Code sample

<details open><summary>Code sample</summary>

```dart
import 'package:flutter/material.dart';

void main() => runApp(MyApp());

class MyApp extends StatelessWidget {
  @override
  Widget build(BuildContext context) {
    return MaterialApp(
      title: 'Flutter Demo',
      home: MyHomePage(),
    );
  }
}

class MyHomePage extends StatefulWidget {
  @override
  _MyHomePageState createState() => _MyHomePageState();
}

class _MyHomePageState extends State<MyHomePage> {
  @override
  Widget build(BuildContext context) {
    return Scaffold(
      appBar: AppBar(
        title: const Text('DatePicker'),
      ),
      body: Center(
        child: Column(
          mainAxisSize: MainAxisSize.min,
          children: [
            ElevatedButton(
              onPressed: () async {
                await showDatePicker(
                  context: context,
                  initialDate: DateTime.now(),
                  firstDate: DateTime(2015),
                  lastDate: DateTime(2050),
                );
              },
              child: const Text('Select date'),
            ),
          ],
        ),
      ),
    );
  }
}
```

</details>

### Flutter Doctor output

<details open><summary>Doctor output</summary>

```console
[✓] Flutter (Channel google3, on macOS 14.2.1 23C71 darwin-x64, locale en)
    • Framework revision 1997bec685 (4 days ago), 2024-01-18T00:00:00.000
    • Engine revision b75d6d80d8
    • Dart version d6c08fa9bb

[✓] Android toolchain - develop for Android devices (Android SDK version Stable)
    • Android SDK at google3
    • Platform Stable, build-tools Stable
    • Java binary at: /Applications/Android Studio with
      Blaze.app/Contents/jbr/Contents/Home/bin/java
    • Java version OpenJDK Runtime Environment (build
      17.0.7+0-17.0.7b1000.6-10550314)

[✓] VS Code (version 1.85.1)
    • VS Code at /Applications/Visual Studio Code.app/Contents
    • Flutter extension version 3.74.0

[✓] Xcode - develop for iOS and macOS (Xcode 15.0)
    • Xcode at /Applications/Xcode.app/Contents/Developer
    • Build 15A240d

[✓] Connected device (1 available)
    • Pixel 4 (mobile) • xx • android-arm64 • Android 13 (API 33)

```

</details>

**Comment by darshankawar** [comment]
I was able to replicate this using latest stable and master versions.

<img width="341" alt="Screenshot 2024-02-02 at 5 59 51 PM" src="https://github.com/flutter/flutter/assets/67046386/f3b74534-cb5b-4918-8e50-f96cf6758f87">

In above image, unable to focus / select on `February 2024` dropdown.

<details>
<summary> stable, master flutter doctor -v </summary>

```
[!] Flutter (Channel stable, 3.16.8, on macOS 12.2.1 21D62 darwin-x64, locale
    en-GB)
    • Flutter version 3.16.8 on channel stable at
      /Users/dhs/documents/fluttersdk/flutter
    ! Warning: `flutter` on your path resolves to
      /Users/dhs/Documents/Fluttersdk/flutter/bin/flutter, which is not inside
      your current Flutter SDK checkout at
      /Users/dhs/documents/fluttersdk/flutter. Consider adding
      /Users/dhs/documents/fluttersdk/flutter/bin to the front of your path.
    ! Warning: `dart` on your path resolves to
      /Users/dhs/Documents/Fluttersdk/flutter/bin/dart, which is not inside your
      current Flutter SDK checkout at /Users/dhs/documents/fluttersdk/flutter.
      Consider adding /Users/dhs/documents/fluttersdk/flutter/bin to the front
      of your path.
    • Upstream repository https://github.com/flutter/flutter.git
    • Framework revision 67457e669f (2 days ago), 2024-01-16 16:22:29 -0800
    • Engine revision 6e2ea58a5c
    • Dart version 3.2.5
    • DevTools version 2.28.5
    • If those were intentional, you can disregard the above warnings; however
      it is recommended to use "git" directly to perform update checks and
      upgrades.

[!] Xcode - develop for iOS and macOS (Xcode 12.3)
    • Xcode at /Applications/Xcode.app/Contents/Developer
    ! Flutter recommends a minimum Xcode version of 13.
      Download the latest version or update via the Mac App Store.
    • CocoaPods version 1.11.2

[✓] Chrome - develop for the web
    • Chrome at /Applications/Google Chrome.app/Contents/MacOS/Google Chrome

[✓] VS Code (version 1.62.0)
    • VS Code at /Applications/Visual Studio Code.app/Contents
    • Flutter extension version 3.21.0

[✓] Connected device (5 available)
    • SM G975F (mobile)       • RZ8M802WY0X • android-arm64   • Android 11 (API 30)
    • Darshan's iphone (mobile)  • 21150b119064aecc249dfcfe05e259197461ce23 •
      ios            • iOS 14.4.1 18D61
    • iPhone 12 Pro Max (mobile) • A5473606-0213-4FD8-BA16-553433949729     •
      ios            • com.apple.CoreSimulator.SimRuntime.iOS-14-3 (simulator)
    • macOS (desktop)            • macos                                    •
      darwin-x64     • Mac OS X 10.15.4 19E2269 darwin-x64
    • Chrome (web)               • chrome                                   •
      web-javascript • Google Chrome 98.0.4758.80

[✓] HTTP Host Availability
    • All required HTTP hosts are available

! Doctor found issues in 1 category.

[!] Flutter (Channel master, 3.19.0-15.0.pre.36, on macOS 12.2.1 21D62
    darwin-x64, locale en-GB)
    • Flutter version 3.19.0-15.0.pre.36 on channel master at
      /Users/dhs/documents/fluttersdk/flutter
    ! Warning: `flutter` on your path resolves to
      /Users/dhs/Documents/Fluttersdk/flutter/bin/flutter, which is not inside
      your current Flutter SDK checkout at
      /Users/dhs/documents/fluttersdk/flutter. Consider adding
      /Users/dhs/documents/fluttersdk/flutter/bin to the front of your path.
    ! Warning: `dart` on your path resolves to
      /Users/dhs/Documents/Fluttersdk/flutter/bin/dart, which is not inside your
      current Flutter SDK checkout at /Users/dhs/documents/fluttersdk/flutter.
      Consider adding /Users/dhs/documents/fluttersdk/flutter/bin to the front
      of your path.
    • Upstream repository https://github.com/flutter/flutter.git
    • Framework revision 31116770ed (2 hours ago), 2024-01-31 19:23:05 -0800
    • Engine revision f4fbabf1eb
    • Dart version 3.4.0 (build 3.4.0-97.0.dev)
    • DevTools version 2.31.0
    • If those were intentional, you can disregard the above warnings; however
      it is recommended to use "git" directly to perform update checks and
      upgrades.

[!] Android toolchain - develop for Android devices (Android SDK version 30.0.3)
    • Android SDK at /Users/dhs/Library/Android/sdk
    ✗ cmdline-tools component is missing
      Run `path/to/sdkmanager --install "cmdline-tools;latest"`
      See https://developer.android.com/studio/command-line for more details.
    ✗ Android license status unknown.
      Run `flutter doctor --android-licenses` to accept the SDK licenses.
      See https://flutter.dev/docs/get-started/install/macos#android-setup for
      more details.

[✓] Xcode - develop for iOS and macOS (Xcode 13.2.1)
    • Xcode at /Applications/Xcode.app/Contents/Developer
    • Build 13C100
    • CocoaPods version 1.11.2

[✓] Chrome - develop for the web
    • Chrome at /Applications/Google Chrome.app/Contents/MacOS/Google Chrome

[✓] IntelliJ IDEA Ultimate Edition (version 2021.3.2)
    • IntelliJ at /Applications/IntelliJ IDEA.app
    • Flutter plugin version 65.1.4
    • Dart plugin version 213.7228

[✓] VS Code (version 1.62.0)
    • VS Code at /Applications/Visual Studio Code.app/Contents
    • Flutter extension version 3.29.0

[✓] Connected device (3 available)
    • Darshan's iphone (mobile) • 21150b119064aecc249dfcfe05e259197461ce23 • ios
      • iOS 15.3.1 19D52
    • macOS (desktop)           • macos                                    •
      darwin-x64     • macOS 12.2.1 21D62 darwin-x64
    • Chrome (web)              • chrome                                   •
      web-javascript • Google Chrome 109.0.5414.119

[✓] Network resources
    • All expected network resources are available.

! Doctor found issues in 1 category.
      
[!] Xcode - develop for iOS and macOS (Xcode 12.3)
    • Xcode at /Applications/Xcode.app/Contents/Developer
    ! Flutter recommends a minimum Xcode version of 13.
      Download the latest version or update via the Mac App Store.
    • CocoaPods version 1.11.2

[✓] Chrome - develop for the web
    • Chrome at /Applications/Google Chrome.app/Contents/MacOS/Google Chrome

[✓] VS Code (version 1.62.0)
    • VS Code at /Applications/Visual Studio Code.app/Contents
    • Flutter extension version 3.21.0

[✓] Connected device (5 available)
    • SM G975F (mobile)       • RZ8M802WY0X • android-arm64   • Android 11 (API 30)
    • Darshan's iphone (mobile)  • 21150b119064aecc249dfcfe05e259197461ce23 •
      ios            • iOS 14.4.1 18D61
    • iPhone 12 Pro Max (mobile) • A5473606-0213-4FD8-BA16-553433949729     •
      ios            • com.apple.CoreSimulator.SimRuntime.iOS-14-3 (simulator)
    • macOS (desktop)            • macos                                    •
      darwin-x64     • Mac OS X 10.15.4 19E2269 darwin-x64
    • Chrome (web)               • chrome                                   •
      web-javascript • Google Chrome 98.0.4758.80

[✓] HTTP Host Availability
    • All required HTTP hosts are available

! Doctor found issues in 1 category.



```

</details>

**Comment by TahaTesser** [comment]
Looks like dropdown is missing semantic node. I can also notice Flutter doesn''t focus on Select Date header as well.

| Google Calendar | Flutter |
| --------------- | --------------- |
| <img src="https://github.com/flutter/flutter/assets/48603081/f46cbb94-2c53-45c9-a947-10012b1ddeec"  /> | <img src="https://github.com/flutter/flutter/assets/48603081/7d7347a2-4b25-4185-ba87-6434d273ff01"  /> |

cc: @HansMuller

**Comment by HansMuller** [comment]
FYI: @MitchellGoodwin

**Comment by hangyujin** [comment]
I have a pr https://github.com/flutter/flutter/pull/143117  to fix the focus on edit field will also fix this issue

**Comment by TahaTesser** [comment]
@hangyujin 
Feel free to take it.

**Comment by jiahaog** [comment]
Changing priority to match internal priority.

**Comment by math1man** [comment]
Can this issue be reopened given the fix was reverted?

**Comment by TahaTesser** [comment]
@math1man Thanks the ping!

**Comment by hangyujin** [comment]
Reopening this as the issue is not fixed in Android Voice Access  and Switch Access 

Android Talkback - Fixed. 
Voice Access - it is unable to select the calendar year ("June 2025") in Voice Access mode.
Switch Access - it is unable to select the calendar year ("June 2025") in Switch Access mode.


iOS
VoiceOver - Fixed.
Voice Control - Fixed.
Switch Control - Fixed.

**Comment by hannah-hyj** [comment]
I cannot reproduce the issue with Voice Access /Switch Access using flutter code example.
Maybe the internal issue is on the app side or related to android versions. 

https://github.com/user-attachments/assets/411d29d5-c6ce-460f-96ee-11a5efd8b0cc


https://github.com/user-attachments/assets/86c712ee-ee71-4bfc-b4fb-d384fe252e6c
